src/cmd/pay.rs

```diff
@@ -125,18 +135,54 @@ fn print_txn(
 pub struct Payee {
     address: PublicKey,
     amount: Hnt,
+    memo: u64,
 }
 
+use crate::result::anyhow;
+
 impl FromStr for Payee {
-    type Err = Box<dyn std::error::Error>;
+    type Err = crate::result::Error;
 
     fn from_str(s: &str) -> std::result::Result<Self, Self::Err> {
-        let pos = s
-            .find('=')
-            .ok_or_else(|| format!("invalid KEY=value: missing `=`  in `{}`", s))?;
-        Ok(Payee {
-            address: s[..pos].parse()?,
-            amount: s[pos + 1..].parse()?,
-        })
+        let mut split = s.split('?');
+
+        if let Some(address) = split.next() {
+            let mut amount = None;
+            let mut memo = 0;
+
+            for segment in split {
+                let pos = segment
+                    .find('=')
+                    .ok_or(|| anyhow!("invalid KEY=value: missing `=`  in `{}`", segment))
+                    .map_err(|_| anyhow!("invalid KEY=value: missing `=`  in `{}`", segment))?;
+
+                let key = &segment[..pos];
+                let value = &segment[pos + 1..];
+                match key {
+                    "amount" => {
+                        amount = Some(value.parse()?);
+                        Ok(())
+                    }
+                    "memo" => {
+                        memo = u64::from_b64(value)?;
+                        Ok(())
+                    }
+                    _ => Err(anyhow!("Invalid key given: {}", key)),
```

Comment:
You'll want to use the `bail!` macro since other wise no error is returned here?